### ISSUE TITLE
What's the recommended approach to use TSOA without class-based controllers?

### ISSUE THREAD

**Issue by nicofraisse** [lukeautry/tsoa#1188]
In our node/express app, we decided to have a "controller-less" architecture. Routes are defined in `[entityName]route.ts` files and handle requests with service functions. What's the recommended approach to use TSOA in such an architecture?

## Sorting

- **I'm submitting a general request/question**

  - [ ] bug report
  - [x] feature request
  - [x] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Possible Solution

I worked around this by creating a class right below my route function each time, giving it an arbitrary name (for example, `UserSwagger` instead of `UserController`, since it technically isn't a controller), and putting all the TSOA code in there. It doesn't feel clean, as the class only exists so that `tsoa spec-and-routes` properly generates the swagger routes. Any suggestions?

**Comment by WoH** [comment]
I'd rephrase this slightly as:

> What's the recommended approach to use TSOA without class-based controllers?

Not using tsoa is the recommended approach here. Fastify and Typebox is what I'd personally recommend if OpenAPI, Validation and TS parity are a concern.

**Comment by DjellalAbdou** [comment]
i have the same question, @nicofraisse did you find a clue ?
